pcweb/pages/docs/component.py

```diff
@@ -74,71 +122,59 @@ def get_props(self) -> list[Prop]:
         if parent_cls != rx.Component and parent_cls != BaseHTML:
             props += Source(component=parent_cls).get_props()
 
-        return props
+        return props 
 
     def _get_props(self) -> list[Prop]:
-        """Get a dictionary of the props and their descriptions.
-
-        Returns:
-            A dictionary of the props and their descriptions.
-        """
-        # The output.
         out = []
-
-        # Get the props for this component.
         props = self.component.get_props()
-
         comments = []
-        # Loop through the source code.
-        for i, line in enumerate(self.code):
-            # Check if we've reached the functions.
-            reached_functions = re.search("def ", line)
-            if reached_functions:
-                # We've reached the functions, so stop.
+
+        i = 0
+        while i < len(self.code):
```

Comment:
why is this change needed - seems like it's logically the same but requires us to manually increment `i` now